File: CalendarApp/app/src/main/java/com/example/calendarapp/MainActivity.java
package com.example.calendarapp;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.GridView;
import android.widget.TextView;

import android.os.Bundle;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import java.util.Locale;

/**
 * 日付選択画面
 */
public class MainActivity extends AppCompatActivity implements AdapterView.OnItemClickListener{

    private TextView titleText;
    private Button prevButton, nextButton;
    private CalendarAdapter mCalendarAdapter;
    private GridView calendarGridView;
    private DateManager dateManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        titleText = findViewById(R.id.titleText);
        prevButton = findViewById(R.id.prevButton);
        prevButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mCalendarAdapter.prevMonth();
                titleText.setText(mCalendarAdapter.getTitle());
            }
        });
        nextButton = findViewById(R.id.nextButton);
        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mCalendarAdapter.nextMonth();
                titleText.setText(mCalendarAdapter.getTitle());
            }
        });
        calendarGridView = findViewById(R.id.calendarGridView);
        mCalendarAdapter = new CalendarAdapter(this);
        calendarGridView.setAdapter(mCalendarAdapter);
        titleText.setText(mCalendarAdapter.getTitle());

        calendarGridView.setOnItemClickListener(this);
    }

    /**
     * カレンダーセルクリック時の処理
     */
    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

        Log.d("position",String.valueOf(position));
        Log.d("date", mCalendarAdapter.getItem(position).toString());

        // グレーアウトセル(カラーコード ： -2302756)をクリック時
        int bkColor = mCalendarAdapter.getBackground(view);
        if(bkColor == -2302756){
            // グレーアウトセルをクリック時のアラートダイアログを作成
            AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
            builder.setMessage("選択された日は予約不可です" );
            builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {

                }
            });
            builder.setCancelable(true);
            builder.show();
        }else {


            //予約日のフォーマットを"MM月dd日(E)"に直す
            SimpleDateFormat dateFormat = new SimpleDateFormat("MM月dd日(E)", Locale.JAPANESE);
            String currentDateString = dateFormat.format(mCalendarAdapter.getItem(position));

            //登録用の予約年月日を取得
            SimpleDateFormat reservationFormat = new SimpleDateFormat("yyyy/MM/dd", Locale.JAPANESE);
            String reservation = reservationFormat.format(mCalendarAdapter.getItem(position));

            // 時間選択画面（TimeActivity）へ遷移
            Intent intent = new Intent(getApplicationContext(), TimeActivity.class);
            intent.putExtra("DATE", currentDateString);
            intent.putExtra("RESERVATION", reservation);

            startActivity(intent);
        }
    }
}

File: MemoKotlin/app/src/main/java/com/example/memokotlin/MainActivity.kt
package com.example.memokotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.*
import kotlinx.android.synthetic.main.activity_create_kotlin_memo.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // ToDo:データベースから値を取得する
        // 仮のデータを作成
        val tmpList = ArrayList<HashMap<String, String>>()
        for (i in 1..5) {
            val data = HashMap<String, String>()
            // 引数には、(名前,実際の値)という組合せで指定します　名前はSimpleAdapterの引数で使用します
            data["body"] = "サンプルデータ$i"
            data["id"] = "sampleId$i"
            tmpList.add(data)
        }

        // Adapter生成
        // ToDo:tmpListを正式なデータと入れ替える
        val simpleAdapter = SimpleAdapter(this,
            tmpList, // 使用するデータ
            android.R.layout.simple_list_item_2, // 使用するレイアウト
            arrayOf("body", "id"), // どの項目を
            intArrayOf(android.R.id.text1, android.R.id.text2) // どのidの項目に入れるか
        )

        // idがmemoListのListViewを取得
        val listView = findViewById<View>(R.id.memoList) as ListView
        listView.adapter = simpleAdapter

        // リスト項目をクリックした時の処理
        listView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            /**
             * @param parent ListView
             * @param view 選択した項目
             * @param position 選択した項目の添え字
             * @param id 選択した項目のID
             */
            // インテント作成  第二引数にはパッケージ名からの指定で、遷移先クラスを指定
            val intent = Intent(this@MainActivity, com.example.memokotlin.CreateKotlinMemo::class.java)

            // 選択されたビューを取得 TwoLineListItemを取得した後、text2の値を取得する
            val two = view as TwoLineListItem
            val idTextView = two.text2 as TextView
            val isStr = idTextView.text as String
            // 値を引き渡す (識別名, 値)の順番で指定します
            intent.putExtra("id", isStr)
            // Activity起動
            startActivity(intent)
        }


        /**
         * 新規作成するボタン処理
         */
        // idがnewButtonのボタンを取得
        val newButton = findViewById<View>(R.id.newButton) as Button
        // clickイベント追加
        newButton.setOnClickListener {
            // CreateMemoActivityへ遷移
            val intent = Intent(this@MainActivity, com.example.memokotlin.CreateKotlinMemo::class.java)
            intent.putExtra("id", "")
            startActivity(intent)
        }
    }
}